test/libsolidity/semanticTests/array/invalid_encoding_for_storage_byte_array.sol

```diff
@@ -0,0 +1,92 @@
+    function arrayPush() public { x.push("t"); }
+    function arrayPushEmpty() public { x.push(); }
+    function del() public { delete x; }
+}
+// ----
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// abiEncode() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// abiEncodePacked() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyToMemory() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// indexAccess() -> 0x6100000000000000000000000000000000000000000000000000000000000000
+// arrayPushEmpty()
+// arrayPush()
+// x() -> 0x20, 5, 0x6162630074000000000000000000000000000000000000000000000000000000
+// arrayPop()
+// assignToLong()
+// x() -> 0x20, 0x25, 0x3132333435363738393031323334353637383930313233343536373839303132, 0x3334353637000000000000000000000000000000000000000000000000000000
+// assignTo()
+// x() -> 0x20, 3, 0x6465660000000000000000000000000000000000000000000000000000000000
+// copyFromStorageShort()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyFromStorageLong()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyToStorage()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// y() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// del()
+// x() -> 0x20, 0x00
+// invalidateXLong()
+// x() -> FAILURE
+// abiEncode() -> FAILURE
+// abiEncodePacked() -> FAILURE
+// copyToMemory() -> FAILURE
+// indexAccess() -> FAILURE
+// arrayPushEmpty()
+// arrayPush()
+// x() -> FAILURE
+// arrayPop()
+// assignToLong() -> FAILURE
+// x() -> FAILURE
+// assignTo() -> FAILURE
+// x() -> FAILURE
+// copyFromStorageShort() -> FAILURE
+// x() -> FAILURE
+// copyFromStorageLong() -> FAILURE
+// x() -> FAILURE
+// copyToStorage() -> FAILURE
+// x() -> FAILURE
+// y() -> 0x20, 0x00
+// del() -> FAILURE
+// x() -> FAILURE
+// invalidateXShort()
+// x() -> FAILURE
+// abiEncode() -> FAILURE
+// abiEncodePacked() -> FAILURE
+// copyToMemory() -> FAILURE
+// indexAccess() -> FAILURE
+// arrayPushEmpty()
+// arrayPush()
+// x() -> FAILURE
+// arrayPop()
+// assignToLong()
+// x() -> 0x20, 0x25, 0x3132333435363738393031323334353637383930313233343536373839303132, 0x3334353637000000000000000000000000000000000000000000000000000000
+// assignTo()
+// x() -> 0x20, 3, 0x6465660000000000000000000000000000000000000000000000000000000000
+// copyFromStorageShort()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyFromStorageLong()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyToStorage()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// y() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
```

Comment:
You should call `invalidateXShort()` here again for this remaining part to actually fulfill its purpose.

Reply:
If the others properly fail ,then this should not be necessary.